File: styleme-flutter/lib/models/prenda_model.dart
// StyleMe - Modelo de datos para Prenda
class PrendaModel {
  final String id;
  final String tipo;
  final String color;
  final String temporada;
  final double confianzaYolo;
  final String imagenUrl;
  final String notas;
  final int vecesUsado;
  final bool activa;
  final String creadoEn;

  PrendaModel({
    required this.id,
    required this.tipo,
    required this.color,
    required this.temporada,
    required this.confianzaYolo,
    required this.imagenUrl,
    this.notas = '',
    this.vecesUsado = 0,
    this.activa = true,
    required this.creadoEn,
  });

  factory PrendaModel.fromJson(Map<String, dynamic> json) {
    return PrendaModel(
      id: json['id']?.toString() ?? '',
      tipo: json['tipo'] ?? '',
      color: json['color'] ?? '',
      temporada: json['temporada'] ?? '',
      confianzaYolo: (json['confianza_yolo'] ?? 0.0).toDouble(),
      imagenUrl: json['imagen_url'] ?? '',
      notas: json['notas'] ?? '',
      vecesUsado: json['veces_usado'] ?? 0,
      activa: json['activa'] ?? true,
      creadoEn: json['creado_en'] ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'tipo': tipo,
        'color': color,
        'temporada': temporada,
        'confianza_yolo': confianzaYolo,
        'imagen_url': imagenUrl,
        'notas': notas,
        'veces_usado': vecesUsado,
        'activa': activa,
        'creado_en': creadoEn,
      };

  // URL completa de la imagen
  String imagenUrlCompleta(String baseUrl) {
    if (imagenUrl.startsWith('http')) return imagenUrl;
    return '$baseUrl$imagenUrl';
  }

  // Porcentaje de confianza formateado
  String get confianzaTexto => '${(confianzaYolo * 100).toStringAsFixed(0)}%';
}

File: styleme-backend/app/models/prenda_model.py
# StyleMe - Modelo de datos para Prenda en MongoDB
from datetime import datetime
from bson import ObjectId
from typing import Optional


class PrendaModel:
    """
    Estructura del documento de prenda en MongoDB.
    Colección: prendas
    """

    @staticmethod
    def crear(
        usuario_id: str,
        tipo: str,
        color: str,
        temporada: str,
        confianza_yolo: float,
        imagen_url: str,
        notas: str = ""
    ) -> dict:
        """Crea un nuevo documento de prenda para insertar en MongoDB."""
        return {
            "usuario_id": ObjectId(usuario_id),
            "tipo": tipo,
            "color": color,
            "temporada": temporada,
            "confianza_yolo": round(confianza_yolo, 4),
            "imagen_url": imagen_url,
            "notas": notas,
            "veces_usado": 0,
            "activa": True,
            "creado_en": datetime.utcnow()
        }

    @staticmethod
    def serializar(doc: dict) -> dict:
        """Convierte un documento MongoDB a dict serializable para JSON."""
        if not doc:
            return {}
        return {
            "id": str(doc["_id"]),
            "usuario_id": str(doc.get("usuario_id", "")),
            "tipo": doc.get("tipo", ""),
            "color": doc.get("color", ""),
            "temporada": doc.get("temporada", ""),
            "confianza_yolo": doc.get("confianza_yolo", 0.0),
            "imagen_url": doc.get("imagen_url", ""),
            "notas": doc.get("notas", ""),
            "veces_usado": doc.get("veces_usado", 0),
            "activa": doc.get("activa", True),
            "creado_en": doc.get("creado_en", datetime.utcnow()).isoformat()
        }
